### PR TITLE
Create features

### PR DESCRIPTION
**Pull request by QIfeiChai**

### CODE REVIEW

File: features [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+T1 = 2019+(31+28)/365
+features['dAccountsOutstandingTime'] = T0-features['dAccountsNextDueDate']
+features['dConfStmtOutstandingTime'] = T0-features['dConfStmtNextDueDate']
+features['dReturnsOutstandingTime'] = T0-features['dReturnsNextDueDate']
+features['OperatingTime'] = T1-features['dIncorporationDate']
+features.loc[features['dAccountsOutstandingTime']<0, 'dAccountsOutstandingTime'] = 0 
+features.loc[features['dConfStmtOutstandingTime']<0, 'dConfStmtOutstandingTime'] = 0 
+features.loc[features['dReturnsOutstandingTime']<0, 'dReturnsOutstandingTime'] = 0
+features.drop(['dAccountsNextDueDate','dConfStmtNextDueDate','dReturnsNextDueDate','dIncorporationDate'],axis = 1,inplace = True)
+feature.head()
```
Comment (Kristina1999) [inline features]: features.head()    ?

Comment (QIfeiChai) [reply features]: Oh that just displays the first five rows of the table. I just wanna have a look at the general format

Comment (Kristina1999) [reply features]: Yeah, I know, sorry, I was commenting on the typo

File: features [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+# In[75]:
+
+
+features.drop(['dConfStmtOutstandingTime','CountryOfOrigin','LimitedPartnershipsNumGenPartners','LimitedPartnershipsNumLimPartners','eCompanyStatus'],axis = 1,inplace = True)
+
+
+# In[76]:
+
+
+features
```
Comment (Kristina1999) [inline features:112]: features = np.array(features)

File: features [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+
+features = df[['CompanyNumber', 'RegAddressPostCode',
+        'CountryOfOrigin','MortgagesNumMortCharges',
+       'MortgagesNumMortOutstanding', 'MortgagesNumMortPartSatisfied', 'LimitedPartnershipsNumGenPartners',
+       'LimitedPartnershipsNumLimPartners',  'SIC1',
+       'eCompanyCategory', 'eCompanyStatus',
+       'eAccountsAccountCategory', 'dIncorporationDate',
+       'dAccountsNextDueDate',
+       'dReturnsNextDueDate','dConfStmtNextDueDate','namechanged', 'namechanged2']].copy()
```
Comment (Kristina1999) [inline features]: features = df[['CompanyNumber', 'oa11',
        'CountryOfOrigin','MortgagesNumMortCharges',
       'MortgagesNumMortOutstanding', 'MortgagesNumMortPartSatisfied', 'LimitedPartnershipsNumGenPartners',
       'LimitedPartnershipsNumLimPartners',  'SIC1',
       'eCompanyCategory', 'eCompanyStatus',
       'eAccountsAccountCategory', 'dIncorporationDate',
       'dAccountsNextDueDate',
       'dReturnsNextDueDate','dConfStmtNextDueDate','namechanged', 'namechanged2', 'class_1']].copy()

Replaced postcode with area code and added class_1 to be extracted later

File: features [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+# In[24]:
+
+
+#keep only those companies whose T0 status is active
+df = df[df['CompanyStatus_T0']=='Active']
+import numpy as np
+#isolate target outcome - company credit status at time T1
+target = np.array(df['class_1'])
```
Comment (Kristina1999) [inline features]: move target assignment to after faulty rows have been removed

**Pull request merged**